### DISCUSSION TITLE
mininotationToTidal

### DISCUSSION THREAD

**Discussion by yaxu** [tidalcycles/Tidal#1164] [Ideas]
It'd be interesting to have a function that could turn a mininotation string into a tidal string that (when evaluated) produced the same pattern.

**Comment by jwaldmann** [comment]
Yes. Related: #217. 

In an ideal world, each mini notation operator (e.g., `[ .. ]` ) has a mini-Syntax AST node (`Stack`) that has a corresponding semantics function (`stack`) that you get by just lower-casing...

To write the translator function (the current issue), all semantics functions that are currently hidden in Sound.Tidal.ParseBP (e.g., `adjust_speed` in `toPat`) will have to be made explicit. This is a good thing.
